routing/fseg.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/limits.hpp"
+
+namespace routing
+{
+class FSegId final
+{
+  static uint32_t constexpr kInvalidFeatureId = numeric_limits<uint32_t>::max();
+
+public:
+  static FSegId MakeInvalid() { return FSegId(kInvalidFeatureId, 0); }
+
+  FSegId() : m_featureId(kInvalidFeatureId), m_segId(0) {}
+
+  FSegId(uint32_t featureId, uint32_t segId) : m_featureId(featureId), m_segId(segId) {}
+
+  uint32_t GetFeatureId() const { return m_featureId; }
```

Comment:
As you're not planning to modify m_featureId, why not to just make them public, but mark as consts? Much less chatty getters.